src/block/tile/BrewingStand.php

```diff
@@ -108,4 +126,125 @@ public function getInventory(){
 	public function getRealInventory(){
 		return $this->inventory;
 	}
+
+	private function checkFuel(Item $item) : void{
+		$ev = new BrewingFuelUseEvent($this);
+		$ev->call();
+
+		if($ev->isCancelled()) {
+			return;
+		}
+
+		$item->pop();
+		$this->inventory->setFuel($item);
+
+		$this->maxFuelTime = $this->remainingFuelTime = $ev->getFuelTime();
+	}
+
+	private function canBrew() : bool{
+		if($this->inventory->getIngredient()->isNull()){
+			return false;
+		}
+
+		for($i = 1; $i <= 3; ++$i){
+			$input = $this->inventory->getItem($i);
+			if($input->isNull()) {
+				continue;
+			}
+
+			if($this->position->getWorld()->getServer()->getCraftingManager()->matchBrewingRecipe($input, $this->inventory->getIngredient()) !== null){
+				return true;
+			}
+		}
+
+		return false;
+	}
+
+	public function onUpdate() : bool{
+		if($this->closed){
+			return false;
+		}
+
+		$this->timings->startTiming();
+
+		$prevBrewTime = $this->brewTime;
+		$prevRemainingFuelTime = $this->remainingFuelTime;
+		$prevMaxFuelTime = $this->maxFuelTime;
+
+		$ret = false;
+
+		$fuel = $this->inventory->getFuel();
+		$ingredient = $this->inventory->getIngredient();
+
+		$canBrew = $this->canBrew();
+
+		if($this->remainingFuelTime <= 0 and $canBrew and $fuel->equals(VanillaItems::BLAZE_POWDER(), true, false)) {
```

Comment:
This blaze powder check should be done inside `checkFuel()` and cause a pre-cancellation of `BrewingFuelUseEvent`, so that plugins can allow different items to be used as fuel.

Reply:
Other items could not have been used as fuel, it is blocked by client.

Reply:
That's beside the point. We don't make API decisions based on what the Mojang client does at a set point in time. The possibility to do this should be there on the server regardless.